Core/URLExtension.swift

```diff
@@ -20,8 +20,15 @@
 
 import Foundation
 
+
+
 extension URL {
     
+    enum URLProtocol: String {
+        case http
+        case https
+    }
+    
     private static let webUrlRegex = "^(https?:\\/\\/)?([\\da-z\\.-]+\\.[a-z\\.]{2,6}|(([\\d]+[.]){3}[\\d]+))([\\/]?[\\/:?=&#]{1}[\\%\\da-zA-Z_\\.-]+)*[\\/\\?]?$"
```

Comment:
you could do it as `(http|https)` instead of https? but I'm not a fan of regex to start with so best to not make it any more complex:

> Person who solve problem with regex, now has two problems. 

;)